lib/grp.gd

```diff
@@ -858,6 +861,10 @@ InstallTrueMethod( IsPolycyclicGroup,
 ##
 DeclareAttribute( "AbelianInvariants", IsGroup );
 
+# minimal number of generators for abelianization. Used internally to check
+# whether it is worth to attempt to reduce generator number
+DeclareAttribute( "AbelianRank", IsGroup );
```

Comment:
[Question] would it be useful to document this attribute?

Reply:
My concern was that this could lead to confusion as `AbelianInvariants` will (in general) return a longer list by separating different primes, while this attribute gives the minimal generator number of $G/G'$.